src/chips/mame_opn2.cpp

```diff
@@ -19,7 +19,8 @@ void MameOPN2::setRate(uint32_t rate, uint32_t clock)
     OPNChipBaseT::setRate(rate, clock);
     if(chip)
         ym2612_shutdown(chip);
-    chip = ym2612_init(NULL, (int)clock, (int)rate, NULL, NULL);
+    uint32_t chipRate = isRunningAtPcmRate() ? rate : nativeRate;
```

Comment:
For best performance experience, check is rate lower than native. When rate is lower - use PCM rate. When is higher - keep native rate. That will don't eat lot of CPU when generating on higher frequencies like 96000 hz. Are you really wanna to tick the chip so strong?

Reply:
This code performs exactly what the function name suggest it does.
I intended to implement this check at higher level, because like I said, this code still needs to have its API implemented to drive it.
I wanted to permit this for testing, because regularly now I switch my audio card between 48k to 88k for tests of multiple configurations.